pkg/ingester/ingester_v2.go

```diff
@@ -1510,10 +1533,10 @@ func (i *Ingester) getOrCreateTSDB(userID string, force bool) (*userTSDB, error)
 	i.userStatesMtx.Lock()
 	defer i.userStatesMtx.Unlock()
 
-	// Check again for DB in the event it was created in-between locks
+	// Check again for DB in the event it was created or closed in-between locks
 	var ok bool
 	db, ok = i.TSDBState.dbs[userID]
-	if ok {
+	if ok && !i.TSDBState.closed {
```

Comment:
Please revert this change. No code using this method should run when ingester is stopping (after adding state-checks to query methods).

Reply:
Good catch, I've removed it.